Chapter_4_Advanced_Structuring_and_Execution.md

```diff
@@ -0,0 +1,599 @@
+- Initializing test data or configurations.
+- Setting the system under test to a known state.
+- Logging into applications or systems.
+- Navigating to the feature under test.
+
+
+
+### 4.1.1 Suite Setup
+
+> [!IMPORTANT]
+> LO-XX Recall key characteristics, benefits, and syntax of Suite Setup (K1)
+>
+> LO-XX Understand when Suite Setup is executed and used (K2)
+
+A **Suite Setup** is executed before any tests|tasks or child suites within the suite are run.
+It is used to prepare the environment or perform actions that need to occur before the entire suite runs.
+Since it is only executed once before all tests|tasks or child suites, it can save time, rather than executing the action for each test|task individually.
+
+**Key characteristics of Suite Setup:**
+- Executed before any tests|tasks and child suites in the suite.
+- If the Suite Setup fails, all tests|tasks in the suite and its child suites are marked as failed, and they are not executed.
+- Ideal for checking **preconditions** that must be met before running the tests|tasks.
+- Logged in the execution log as a separate section, indicating the setup status.
+
+**Typical use cases:**
+- Ensuring that the environment is ready for execution.
+- Starting services or applications required for the suite.
+- Preparing a system under test to meet the suite's requirements.
+- Loading configurations or resources shared across multiple tests|tasks.
+
+Example of defining a Suite Setup:
+
+```robotframework
+*** Settings ***
+Suite Setup    Initialize Environment   dataset=Config_C3
+```
+
+
+
+### 4.1.2 Test|Task Setup
+
+> [!IMPORTANT]
+> LO-XX Recall key characteristics, benefits, and syntax of Test Setup (K1)
+>
+> LO-XX Understand when Test|Task Setup is executed and used (K2)
+
+A **Test|Task Setup** is executed before a single test|task runs.
+It is used to prepare the specific conditions required for that test|task.
+
+You can define a default Test|Task Setup in the `*** Settings ***` section of the suite using the `Test Setup`|`Task Setup` setting.
+This setup will be applied to all tests|tasks within the suite unless overridden.
+
+Individual tests|tasks can override the default setup by specifying their own `[Setup]` setting within the test|task.
+To disable the setup for a specific test|task, you can set `[Setup]    NONE`, which means that no setup will be executed for that test|task.
+
+**Key characteristics of Test|Task Setup:**
+- Executed before the test|task starts.
+- If the Test|Task Setup fails, the test|task is marked as failed, and its body, including its main keywords, is not executed.
+- Can be set globally for all tests|tasks in a suite and overridden locally.
+- Logged in the execution log as a separate section, indicating the setup status.
+
+**Typical use cases:**
+- Setting up test data unique to the test|task.
+- Distinguishing keywords of a test.
+- Executing preparation steps to navigate to the feature under test.
```

Comment:
Don't make this test specific, because it could be the same for tasks:

```suggestion
- Execute preparation steps to navigatie to the test feature/task
```

Reply:
No i disagree with the grammar.
And here i think this is kind of specific to testing.
@frankvanderkuur

Reply:
Grammar-wise I'm fine with whatever solution. Although I don't have heaps of experience with RPA, I can see situations where you could benefit from using a task setup. But, I wouldn't say this needs a whole lot of debating. In or out, either is fine with me.

Reply:
changed